docs/tutorial.md

```diff
@@ -878,6 +880,51 @@ local function do_something(p: MyPoint)
    -- ...
 end
 ```
+## Integers and numbers
+
+The integer sub-type was added with release 0.13.0. It is of undefined
+precision, deferring to the Lua VM.
+
+* integer constants without a dot (e.g. 123) are of type integer
+* operations preserve the integer type according to Lua 5.4 semantics (e.g.
+  `integer + integer = integer`, `integer + number = number`, etc.)
+* `integer` is a subtype of `number`
+   * this generally means that an `integer` is accepted everywhere a `number` is
+     expected
+   * note that this does not extend to type variables in generics, which are
+     invariant: `Foo<integer>` is not the same as `Foo<number>`
+   * if a variable is of type `integer`, then an arbitrary `number` is not
+     accepted
+     * `local x: integer = 1.0` doesn't work, even though the floating-point
+       number has a valid integer representation
+* variables (and type variables) are inferred as integer when initialized with
+  integers
+  * this can happen directly (`local x = 1`) or indirectly (`local x =
+    math.min(i, j)`)
+  * to infer an integral value as a `number`, use .0: `local x = 1.0` infers a
+    `number`
+
+### Integers and the standard library
```

Comment:
These notes about the standard library from the release notes were mainly aimed at Teal programmers who were migrating from the version without integers. Given the lenient behavior of the standard library definitions regarding numbers and integers, I don't expect newcomers to have any trouble with them, since things generally work as one would expect (at least, nothing deserving a full section in the tutorial!) — do you have any specific concerns about surprising behavior in the standard library that you feel need to be called out in the docs?